src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -350,10 +356,51 @@ public override void VisitMarkupTransition(MarkupTransitionSyntax node)
         }
         #endregion Razor
 
+        private static int GetElementSemanticKind(SyntaxNode node)
+        {
+            var semanticKind = IsComponent(node) ? RazorSemanticTokensLegend.RazorComponentElement : RazorSemanticTokensLegend.RazorTagHelperElement;
+            return semanticKind;
+        }
+
+        private static int GetAttributeSemanticKind(SyntaxNode node)
+        {
+            var semanticKind = IsComponent(node) ? RazorSemanticTokensLegend.RazorComponentAttribute : RazorSemanticTokensLegend.RazorTagHelperAttribute;
+            return semanticKind;
+        }
+
+        private static bool IsComponent(SyntaxNode node)
+        {
+            if (node is MarkupTagHelperElementSyntax element)
+            {
+                var componentDescriptor = element.TagHelperInfo.BindingResult.Descriptors.FirstOrDefault(d => d.IsComponentTagHelper());
```

Comment:
Is I messed this up at all I assume it will be right here. I'm not 100% confident that this is the right way to determine if it is a component or not because I don't know much about our Descriptors, so correct me if I'm off.

Reply:
I think this is right. @pranavkm for confirmation